src/ObjLoading/Game/T6/BSP/Linker/BSPLinker.cpp

```diff
@@ -0,0 +1,106 @@
+#include "BSPLinker.h"
+
+#include "ClipMapLinker.h"
+#include "ComWorldLinker.h"
+#include "GameWorldMpLinker.h"
+#include "GfxWorldLinker.h"
+#include "MapEntsLinker.h"
+#include "SkinnedVertsLinker.h"
+
+namespace BSP
+{
+    FootstepTableDef* BSPLinker::addEmptyFootstepTableAsset(std::string assetName)
+    {
+        if (assetName.length() == 0)
+            return nullptr;
+
+        FootstepTableDef* footstepTable = m_memory.Alloc<FootstepTableDef>();
+        footstepTable->name = m_memory.Dup(assetName.c_str());
+        memset(footstepTable->sndAliasTable, 0, sizeof(footstepTable->sndAliasTable));
+
+        m_context.AddAsset<AssetFootstepTable>(assetName, footstepTable);
+
+        return footstepTable;
+    }
+
+    bool BSPLinker::addDefaultRequiredAssets(BSPData* bsp)
+    {
+        if (m_context.LoadDependency<AssetScript>("maps/mp/" + bsp->name + ".gsc") == nullptr)
+            return false;
+        if (m_context.LoadDependency<AssetScript>("maps/mp/" + bsp->name + "_amb.gsc") == nullptr)
+            return false;
+        if (m_context.LoadDependency<AssetScript>("maps/mp/" + bsp->name + "_fx.gsc") == nullptr)
+            return false;
+
+        if (m_context.LoadDependency<AssetScript>("clientscripts/mp/" + bsp->name + ".csc") == nullptr)
+            return false;
+        if (m_context.LoadDependency<AssetScript>("clientscripts/mp/" + bsp->name + "_amb.csc") == nullptr)
+            return false;
+        if (m_context.LoadDependency<AssetScript>("clientscripts/mp/" + bsp->name + "_fx.csc") == nullptr)
+            return false;
```

Comment:
nit: the codebase generally tries to use std::format instead of string concatination

adjusted in: 5e06df6286f30de54027f115c08c94e986bfe73c